ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Query.hs

```diff
@@ -575,3 +663,50 @@ decodeShelleyResult query = case query of
     GetStakePools                              -> fromCBOR
     GetStakePoolParams {}                      -> fromCBOR
     GetRewardInfoPools                         -> fromCBOR
+    GetPoolState {}                            -> fromCBOR
+    GetStakeSnapshot {}                        -> fromCBOR
+
+data StakeSnapshot crypto = StakeSnapshot
```

Comment:
Please add a Haddock comment explaining what this is.

Also, I think these fields shouldn't be `Integer`s, but rather `SL.Coin`?

Reply:
Why not return `SnapShot crypto`? It seems like we are losing useful information i.e the delegations and pool params. The consumer can do these calculations (indeed we can expose functions from `cardano-api` to do this)

Reply:
It is define like this:

```
data SnapShots crypto = SnapShots
  { _pstakeMark :: SnapShot crypto, -- Lazy on purpose
    _pstakeSet :: !(SnapShot crypto),
    _pstakeGo :: !(SnapShot crypto),
    _feeSS :: !Coin
  }
  deriving (Show, Eq, Generic)

data SnapShot crypto = SnapShot
  { _stake :: !(Stake crypto),
    _delegations :: !(VMap VB VB (Credential 'Staking crypto) (KeyHash 'StakePool crypto)),
    _poolParams :: !(VMap VB VB (KeyHash 'StakePool crypto) (PoolParams crypto))
  }
  deriving (Show, Eq, Generic)
```

The maps stored are very big, so in the interest of improving memory and cpu efficiency and not transferring more data than necessary over the socket, I do the map lookup on the node side.

Reply:
> Why not return `SnapShots crypto`?

> The maps stored are very big [..., so] I do the map lookup on the node side

On Slack, Jordan also asked: "Should we allow the possibility to get the stake snapshot for multiple pools at once?"

One option occurs to me: we return a `SnapShots crypto` (so we don't have to define a new type 🙌) but we restrict all the maps inside that data type to just the keys given (so we support a multi-pool query 🙌).

-----

On the slack thread, you agreed that one-pool at a time is fine for the first PR. So that's not urgent.

But avoiding a new data type might be nice.

So: does the resuse of `SnapShots crypto` but with all the maps restricted down to the pools identified by the query make sense?

One immediate downside: such a restricted `SnapShot` is probably counter-intuitive: eg it violates some generally-expected invariants of `SnapShot` (along the lines of _completeness_)